src/components/inputs/PublishButton/PublishButton.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Event } from 'src/types/domain';
+import { updateEvent } from 'src/api/events';
+import { EventStatus } from 'src/utils/constants';
+import Button from 'src/components/inputs/Button';
+import Modal from 'src/components/Modal';
+import ConfirmationModal from 'src/components/Modal/ConfirmationModal';
+import { useModal } from 'src/hooks/useModal';
+
+type PublishButtonProps = {
+  event: Event;
+  onPublish: (path: string) => void;
+};
+
+function PublishButton({ event, onPublish }: PublishButtonProps) {
+  const { isShown, toggle } = useModal();
+
+  const onConfirm = () => {
+    handlePublish(event);
+    toggle();
+  };
+  const onCancel = () => toggle();
+
+  const handlePublish = async (event: Event) => {
+    await updateEvent(
+      event.id,
+      event.startTime,
+      event.endTime,
+      event.name,
+      event.description,
+      event.eventType,
+      event.imageUrl,
+      event.imageAlt,
+      event.location,
+      EventStatus.PUBLISHED,
+      event.teamsUrl
+    );
+    onPublish(`/event/${event.id}/${event.name}`);
+  };
```

Comment:
Not quite the same situation, but I just wanted to mention that this type of naming collision is why I always prefer to:

1. use `handle` as prefix for "event handlers" created inside of a component; and
2. use `on` as prefix for "event handlers" received as props

In that way, there will never be a naming collision, even if you need to wrap the event handler prop call inside of another function to include some additional logic.